src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -295,6 +297,24 @@ private ZipFileProcessingResult processZipFileContent(
         } catch (PreviouslyFailedToUploadException ex) {
             log.warn("Rejected file {} from container {} - failed previously", zipFilename, containerName, ex);
             handleEventRelatedError(Event.DOC_UPLOAD_FAILURE, containerName, zipFilename, ex);
+            return null;
+        } catch (DataIntegrityViolationException ex) {
+            // only report on constraint violations
+            if (ex.getCause() instanceof ConstraintViolationException) {
+                ConstraintViolationException exception = (ConstraintViolationException) ex.getCause();
+                log.warn(
+                    "Rejected file {} from container {} - DB constraint violation {}",
+                    zipFilename,
+                    containerName,
+                    exception.getConstraintName(),
+                    exception
+                );
+                handleInvalidFileError(Event.FILE_VALIDATION_FAILURE, containerName, zipFilename, leaseId, exception);
```

Comment:
I think we should be very strict here and rely on the DNC-specific constraint to conclude the file is invalid. Otherwise, it's a file processing error (to be investigated by us) which shouldn't result in rejection.

Reply:
Let's organise a chat as this conversation might get stuck or go on forwards and backwards 😕

Reply:
Or can introduce extra condition like:
```java
if (exception.getConstraintName().equals("scannable_item_dcn")) {
    handleInvalidFileError(Event.FILE_VALIDATION_FAILURE, containerName, zipFilename, leaseId, exception);
} else {
    handleEventRelatedError(Event.DOC_FAILURE, containerName, zipFilename, exception);
}
```
would that be sufficient?

Reply:
Fine by me